File: leetcode0169/solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        c = Counter(nums)
        l = len(nums) // 2

        for k, v in c.items():
            if v > l:
                return k

# class Solution:
#     def majorityElement(self, nums: List[int]) -> int:
#         if len(nums) == 1:
#             return nums[0] 
#         nums.sort()
#         count = 1
#         for i in range(len(nums) - 1):
#             if nums[i] == nums[i + 1]:
#                 count += 1
#                 if count > len(nums) // 2:
#                     return nums[i]
#             else:
#                 count = 1
            

        

File: leetcode0169/solution.rs
impl Solution {
    pub fn majority_element(mut nums: Vec<i32>) -> i32 {
        if nums.len() == 1 {
            return nums[0];
        }
        nums.sort_unstable();

        let mut count: i32 = 1;

        for i in 1..nums.len() {
            if nums[i] == nums[i - 1] {
                count += 1;
            } else {
                count = 1;
            }

            if count > (nums.len() as i32) / 2 {
                return nums[i];
            }
        }

        -1
    }
}